dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -2279,6 +2277,29 @@ void Sedp::process_discovered_writer_data(DCPS::MessageId message_id,
       }
       match_endpoints(guid, top_it->second);
     } else {
+
+#if OPENDDS_CONFIG_OWNERSHIP_KIND_EXCLUSIVE
+      if (iter->second.writer_data_.ddsPublicationData.ownership.kind == DDS::EXCLUSIVE_OWNERSHIP_QOS &&
+          wdata.ddsPublicationData.ownership.kind == DDS::EXCLUSIVE_OWNERSHIP_QOS &&
+          iter->second.writer_data_.ddsPublicationData.ownership_strength.value != wdata.ddsPublicationData.ownership_strength.value) {
+
+        topic_name = iter->second.get_topic_name();
+        TopicDetailsMap::const_iterator top_it = topics_.find(topic_name);
+        using DCPS::RepoIdSet;
+        const RepoIdSet& assoc =
+          (top_it == topics_.end()) ? RepoIdSet() : top_it->second.local_subscriptions();
+        for (RepoIdSet::const_iterator i = assoc.begin(); i != assoc.end(); ++i) {
+          LocalSubscriptionIter lsi = local_subscriptions_.find(*i);
+          if (lsi != local_subscriptions_.end()) {
+            DCPS::DataReaderCallbacks_rch sl = lsi->second.subscription_.lock();
+            if (sl) {
+              sl->update_ownership_strength(guid, wdata.ddsPublicationData.ownership_strength.value);
```

Comment:
I haven't looked through all of this yet - just wanted to note that we should verify that any locks held here are safe to have during the upcall.